lib/streamlit/runtime/scriptrunner/exec_code.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable
+
+from streamlit.error_util import handle_uncaught_app_exception
+from streamlit.runtime.scriptrunner.exceptions import RerunException, StopException
+
+if TYPE_CHECKING:
+    from streamlit.runtime.scriptrunner.script_requests import RerunData
+    from streamlit.runtime.scriptrunner.script_run_context import ScriptRunContext
+
+
+def wrap_in_try_and_exec(
```

Comment:
Hm, I don't currently have any good suggestions for names for this function, but I think the current name may be a bit too general. Not the biggest deal, but it's probably worth spending some time trying to think of a more descriptive name.

Reply:
yeah I agree! What about something like: `exec_func_with_error_handling` or similar?